fdbserver/DataDistribution.actor.cpp

```diff
@@ -3106,7 +3110,29 @@ ACTOR Future<Void> teamTracker(DDTeamCollection* self, Reference<TCTeamInfo> tea
 			ASSERT( self->optimalTeamCount >= 0 );
 			self->zeroOptimalTeams.set(self->optimalTeamCount == 0);
 		}
-		throw;
+		state bool sendShardsPath = false;
+		try {
+			// If server is marked as failed, a lost race condition could result in RelocateShards not being sent
+			// This is implemented as a safety net so we do not have orphaned shards sitting around
+			if (e.code() == error_code_actor_cancelled && teamContainsFailedServer(self, team)) {
+				TraceEvent("TeamTrackerStoppedSendRelocateToDDQ", self->distributorId).detail("Team", team->getDesc());
+				sendShardsPath = true;
+				teamTrackerSendRelocateShards(self, team, redundantTeam, teamContainsFailedServer(self, team), serversLeft);
+			}
+			// Trace the type of error that stopped the tracker, even if it is actor_cancelled
+			if (logTeamEvents) {
+				TraceEvent("TeamTrackerStopped", self->distributorId).error(e, true);
+			}
+			throw e;
+		} catch (Error& err) {
+			// This should only be logged as an error if an error was thrown due to the sending of
+			// RelocateShard requests. This avoids falsely logging an error on rethrows with
+			// actor_cancelled, since .error will not log actor_cancelled by default.
+			if (sendShardsPath) {
```

Comment:
If there is no error thrown,  `sendShardsPath `  will still be set to true and the SevError will be triggered.
`sendShardsPath ` should be set false before `throw e`. 
The name `sendShardsPath ` can be changed to `hasErrorInSendShards`, which is more intuitive IMHO.

Reply:
The SevError will not be triggered because the error code in that case will be actor_cancelled, which does not get logged by default (you must call `.error(e, true)` for it to be logged). Would you still prefer your suggested change for clarity purposes though? I'm just trying to avoid any extra operations.